### PR TITLE
Update php-definitions.md

### PR DESCRIPTION
**Pull request by mindplay-dk**
Improve documentation for DI\factory()

### CODE REVIEW

File: doc/php-definitions.md [hunk 1]
```diff
@@ -111,24 +111,41 @@ return [
...
+needed - assuming you have auto-wiring enabled, or have already registered a
+definition for `FooFactory`, you can use this definition:
 
+```php
+return [
+    Foo::class => DI\factory([DI\get(FooFactory::class), 'create']),
+];
+```
+
+Or, if `FooFactory::create()` were a static method, you could use this definition:
```
Comment (mindplay-dk) [inline doc/php-definitions.md]: Is the static example really necessary or too verbose? Should we drop it?

File: doc/php-definitions.md [hunk 2]
```diff
@@ -111,24 +111,41 @@ return [
...
+];
+```
+
+It's usually better to lazy-load `FooFactory` and let the container create it as
+needed - assuming you have auto-wiring enabled, or have already registered a
+definition for `FooFactory`, you can use this definition:
 
+```php
+return [
+    Foo::class => DI\factory([DI\get(FooFactory::class), 'create']),
```
Comment (mindplay-dk) [inline doc/php-definitions.md]: Is this example correct? Will DI\factory() detect and resolve the reference created by `DI\get(FooFactory::class)` inside the callable? (if so, this is not strictly a callable, is it?)

**Comment by mindplay-dk** [comment]
@mnapoli see my comments inline above.

File: doc/php-definitions.md (hunk 2 above)
Comment (mnapoli) [inline doc/php-definitions.md]: Not yet! See #285 (yet another reason to use `Container::call()` to call factories maybe ;)

Comment (mnapoli) [inline doc/php-definitions.md]: > (if so, this is not strictly a callable, is it?)

not strictly a callable indeed

File: doc/php-definitions.md (hunk 1 above)
Comment (mnapoli) [inline doc/php-definitions.md]: I think it's the only example we can use for now since #285 isn't implemented.

Comment (mindplay-dk) [inline doc/php-definitions.md]: [Removed](https://github.com/mindplay-dk/PHP-DI/commit/671197e3014e13b95df2a4609def037a044643d7) with a reference to #285 for future rollback.

**Comment by mnapoli** [comment]
Perfect thank you!

**Pull request merged**